app/src/main/assets/native-10.7/ExternalPlayerPlugin.js

```diff
@@ -49,6 +49,9 @@ export class ExternalPlayerPlugin {
         this._currentTime = (options.playerStartPositionTicks || 0) / 10000;
         this._paused = false;
         this._currentSrc = options.url;
+        this._isIntro = typeof options.item != "undefined" &&
+            typeof options.item.ProviderIds != "undefined" &&
+            options.item.ProviderIds.hasOwnProperty("prerolls.video");
```

Comment:
No JS wizard, but couldn't this be simplified a bit? Does it really need the `typeof` here?

Reply:
This is the "safe" approach

I found something (Kotlin?), although it is relatively new, and some old webview may not be compatible.

https://github.com/tc39/proposal-optional-chaining
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

`options?.item?.ProviderIds?.["prerolls.video"] != undefined`

More options, choose one if you like: https://stackoverflow.com/questions/2631001/test-for-existence-of-nested-javascript-object-key

Or ask for your nearest JS Wizard (?)

Reply:
Ex-JS Wizard here 🧙 

I think we could use JavaScript quirks to simplify to:

```suggestion
        this._isIntro = options.item && options.item.ProviderIds && options.item.ProviderIds.hasOwnProperty('prerolls.video');
```

Basically we check if the members are true-ish, `undefined` or `null` is falsy so this check makes sure  the members exists before calling the hasOwnProperty function.

Reply:
Ah yes, the magic world of JS, that worked before 

https://github.com/jellyfin/jellyfin-android/blob/6cc40f6571f554859446ad35cb4e58138a0d492d/app/src/main/assets/native-10.7/ExternalPlayerPlugin.js#L36-L37

Anything else?

Reply:
Nope, should be fine now.